### ISSUE TITLE
[Request] Instructions to build appimage or normal installation package or pip install

### ISSUE THREAD

**Issue by omega3** [Elleo/pied#6]
I am on Manjaro Linux and I don't know how to build appimage. Can you provide some instructions how to build appimage or normal package that installs like all other packages. I prefer smaller packages than all this Flatpak ecosystem. 
Does it need to be run in some virtual environment like env or something?

Or perhaps pip install? I installed piper via pip
`pip install piper-tts`
Pip would be the best because is so easy to install.

**Comment by Elleo** [comment]
Hi,

Unfortunately the AppImage packaging isn't ready yet (I'm running into a crash when packaged as an AppImage that I haven't been able to track down yet).

There is a pre-compiled binary version here: https://github.com/Elleo/pied/releases/download/v0.2.0/pied-0.2.0-x86_64.tar.gz

No guarantees that it'll work on your system though, as versions of the libraries that it depends on (like gstreamer) may be different/absent. But it's probably worth a try.

I'm afraid pip wouldn't be appropriate as Pied isn't python based.

I'll leave this bug open until the AppImage is complete.

Hope the helps,
Mike.

**Comment by omega3** [comment]
Thank you very much. It works for me. I run it even without installing:
`./pied`
I am very happy with that. Great software.

Sorry for offtopic but are there any chances to add some options, to be able to choose for example en-US libritts_r **1638 (7)**?
https://rhasspy.github.io/piper-samples/

**Comment by Elleo** [comment]
@omega3 Yep, I'm planning on adding support for models containing multiple voices in a future version

**Comment by Elleo** [comment]
Hi @RobertsGhostBusterFan,

I'm afraid a Windows port wouldn't really be feasible. Pied is based around the speech dispatcher architecture which is a Linux (and BSD) specific system. Windows has its own, completely different text-to-speech system. So, for it to work on Windows would require a complete rewrite targetting the Windows system (which I don't have any experience or knowledge of).

Cheers,
Mike

**Comment by raininja** [comment]
I am here to also request normal building instructions, as I prefer not to use appimage nor flatpak in my arch installation. I have attempted and dart analyze yields numerous errors . . .

**Comment by raininja** [comment]
GOT IT! I feel dumb , it's a flutter app, use flutter methods.

**Comment by juanMarinero** [comment]
Hi! I built [Piper from source](https://github.com/rhasspy/piper?tab=readme-ov-file#installation) in next path (added to ~/.bashrc)

`export PATH=$PATH:$HOME/piper/install`

So I can run "piper" anywhere instead of cd to repo dir and "./piper". Example:
```
cd [anyPath]
echo 'Welcome to the world of speech synthesis!' | \
  piper --model en_US-lessac-medium.onnx --output_file welcome.wav
```

I linked that executable to a location I expect snap to have access (I think this is **my error**, but no idea where to link)
`sudo ln -s $HOME/piper/install/piper /usr/local/bin/piper`

Then I snap installed Pied 
```
sudo snap install pied
pied 
```
Piped **asks to install Piper**. Though its installed and linked to /usr/local/bin

With newest pre-compiled binary version of Piped happens the same. With flutter no idea, but it is unfamiliar to me.

FYI:
If I install the "Piper" through Piped GUI and run as I explained in [issue 14](https://github.com/Elleo/pied/issues/14#issuecomment-2396805603) it works. But I insist, I want Piped to find git-cloned "Piper" please

More info:

**Ubuntu 24.04**.1 LTS

<pre><font color="#5FD700">&gt;</font> <font color="#26A269">snap</font> list pied
Name  Version  Rev  Tracking       Publisher        Notes
pied  0.2.1    82   latest/stable  michael-sheldon  -</pre>


<pre><font color="#5FD700">&gt;</font> <font color="#26A269">snap</font> connections pied
Interface               Plug                               Slot                            Notes
audio-playback          pied:audio-playback                :audio-playback                 -
content[gnome-42-2204]  pied:gnome-42-2204                 gnome-42-2204:gnome-42-2204     -
content[gtk-3-themes]   pied:gtk-3-themes                  gtk-common-themes:gtk-3-themes  -
content[icon-themes]    pied:icon-themes                   gtk-common-themes:icon-themes   -
content[sound-themes]   pied:sound-themes                  gtk-common-themes:sound-themes  -
desktop                 pied:desktop                       :desktop                        -
desktop-legacy          pied:desktop-legacy                :desktop-legacy                 -
gsettings               pied:gsettings                     :gsettings                      -
network                 pied:network                       :network                        -
network-status          pied:network-status                :network-status                 -
opengl                  pied:opengl                        :opengl                         -
personal-files          pied:dot-config-speech-dispatcher  :personal-files                 -
process-control         pied:process-control               :process-control                -
wayland                 pied:wayland                       :wayland                        -
x11                     pied:x11                           :x11                            -
</pre>
